examples/include/crypto-accelerator.p4

```diff
@@ -122,6 +128,29 @@ extern crypto_accelerator {
     void enable_encrypt<T>(in T enable_auth);
     void enable_decrypt<T>(in T enable_auth);
 
+    // crypto accelerator runs immediately and returns control flow to the current pipeline
+    // stage. The method is responsible for creating the ESP header, calculating the payload 
+    // offset and lengths, and then encrypting the payload appropriately and reparsing the 
+    // packet.
+    // H - inout Headers that is the output of the parser block
+    // M - inout Metadata that is from the parser block and shared with the control 
+    crypto_results_e encrypt_inline<H,M,T,S,I>(inout H hdr, inout M meta,
```

Comment:
add a comment of order of operations. e.g. deparser or byte stream update must occur when the extern is invoked

Reply:
add comment on preconditions and postconditions

Reply:
add side effects

Reply:
it may need packet in as well